### PR TITLE
Add Validators. Refactor coreSDK folder

### PR DESCRIPTION
**Pull request by barustum**
- Add Validators
- Move files from coreSDK to Common and Channel
- Fix cjs and amd folder structures for Common and Channel to match coreSDK

### CODE REVIEW

**Comment by barustum** [comment]
Please note that the files under "Generated" are modified manually and not generated. I will investigate how to run the generation tool and update them.

File: coreSDK/amd/.npmignore [hunk 1]
```diff
@@ -5,6 +5,6 @@
 
 # ... but these files
 !package.json
-!bundle/aicore.js
```
Comment (barustum) [inline coreSDK/amd/.npmignore]: @jpiyali I changed from aicore to applicationinsights-core-js for consistency. I can change it back if you feel like we should stick with aicore. I can go either way. If we change it to applicationinsights-core-js, we'll need to update the npm-vsts feed

Comment (jpiyali) [reply coreSDK/amd/.npmignore]: should the file name be just core.js and not so large? I checked a few other existing public packages, names do not need to be reflective of package?

File: AppInsightsChannel/Sender.ts [hunk 2]
```diff
@@ -13,31 +14,43 @@ import { MetricValidator } from './TelemetryValidation/MetricValidator';
...
 declare var XDomainRequest: {
     prototype: IXDomainRequest;
     new(): IXDomainRequest;
 };
 
-export class Sender implements ITelemetryPlugin {
-    priority: number; // todo set priority
-    public identifier: string; // todo set identifier
+export class Sender implements IChannelControls {
+    public priority: number; // todo set priority
```
Comment (jpiyali) [inline AppInsightsChannel/Sender.ts]: set any value > 100

File: AppInsightsChannel/TelemetryValidation/PageViewPerformanceValidator.ts [hunk 3]
```diff
@@ -1,10 +1,28 @@
...
+        if (!item.domainProperties ||
+            !item.domainProperties["domProcessing"] ||
+            !item.domainProperties["duration"] ||
+            !item.domainProperties["name"] ||
+            !item.domainProperties["networkConnect"] ||
+            !item.domainProperties["perfTotal"] ||
+            !item.domainProperties["receivedResponse"] ||
+            !item.domainProperties["sentRequest"] ||
+            !item.domainProperties["url"]) {
+            return false;
```
Comment (jpiyali) [inline AppInsightsChannel/TelemetryValidation/PageViewPerformanceValidator.ts]: add log when returning false.

File: AppInsightsChannel/cjs/package.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+    "name": "applicationinsights-channel-js",
+    "version": "0.0.1",
+    "description": "Microsoft Application Insights JavaScript SDK Channel",
+    "main": "bundle/aichannel.js",
+    "devDependencies": {
+        "grunt": "1.0.1",
+        "grunt-contrib-qunit": "2.0.0",
```
Comment (jpiyali) [inline AppInsightsChannel/cjs/package.json]: we should review each of our new package.json for the grunt dependency.

**Pull request merged**